components/EasySPA.php

```diff
@@ -21,7 +32,7 @@ public function onRun()
             $this->addJs(['js/statemanager.js', 'js/assetmanager.js', 'js/easyspa.js']);
 
             $this->controller->bindEvent('page.render', function ($contents) {
-                return '<div id="easyspa-container">' . $contents . '</div>';
+                return $contents;
```

Comment:
When i leave it all it does it wrap the element twice so it would look like 
```
<div id="easyspa-container">
<div id="easyspa-container">
{ content }
</div>
</div>
```
which messes with grid contented that i use

Reply:
The problem here, is `Request::header('X_OCTOBER_REQUEST_HANDLER')` will never exists on `onRun` method, only exists when call `init()` method. This is why element `#easyspa-container` has returned in `onGetPage()` method.

Reply:
I solve this issue with this:
```php
    public function onRun()
    {
        // Do required initial work only if this isn't a request to load a page
        if (!Request::hasHeader('X_OCTOBER_REQUEST_PARTIALS') && Request::header('X_OCTOBER_REQUEST_HANDLER') !== 'onGetPage') {
            $this->assetPath = plugins_path('luketowers/easyspa/assets');
            $this->addJs(['js/statemanager.js', 'js/assetmanager.js', 'js/easyspa.js']);

            $this->controller->bindEvent('page.render', function ($contents) {
                return '<div id="easyspa-container">' . $contents . '</div>';
            });
        }
    }
```

Reply:
This would still end up doubling up with ForceUrlSchema